tests/compiler/std/array.ts

```diff
@@ -978,5 +978,30 @@ class Ref { constructor() {} }
   assert(subarrU32.toString() == "1");
 }
 
+// Array#flat //////////////////////////////////////////////////////////////////////////////////
+{
+  var flatTarget: i32[][] = [[0], [1, 2, 3], [4, 5, 6], [7, 8, 9]];
+  var result = flatTarget.flat();
+  for (let i = 0; i < 10; i++) {
+    assert(result[i] == i, (i).toString());
+  }
+  assert(result.length == 10);
+
+  var flatStringTarget: string[][] = [["one"], ["two", "three"], ["four", "five", "six"], null, ["seven"]];
```

Comment:
what about?
```ts
[["one"], ["two", "three"], ["four", null, "six"], null, ["seven"]];
```
Just added `null` for internal array for better coverage

Reply:
That sounds right. Is this all set?